app/actions/websocket/category.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {deleteCategory, storeCategories} from '@actions/local/category';
+import {fetchCategories} from '@actions/remote/category';
+import DatabaseManager from '@database/manager';
+import {queryCategoriesById} from '@queries/servers/categories';
+
+type WebsocketMessage = {
+    broadcast: {
+        team_id: string;
+    };
+    data: {
+        team_id: string;
+        category?: string;
+        category_id: string;
+        updatedCategories?: string;
+        order?: string[];
+    };
+}
+
+const addOrUpdateCategories = async (serverUrl: string, categories: CategoryWithChannels[]) => {
+    try {
+        storeCategories(serverUrl, categories);
+    } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log('Category WS: addOrUpdateCategories', e, categories);
+    }
+};
+
+export async function handleCategoryCreatedEvent(serverUrl: string, msg: WebsocketMessage) {
+    let category;
+    try {
+        category = JSON.parse(msg.data.category!);
+        addOrUpdateCategories(serverUrl, [category]);
+    } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log('Category WS: handleCategoryCreatedEvent', e, msg);
+        fetchCategories(serverUrl, msg.broadcast.team_id);
+    }
+}
+
+export async function handleCategoryUpdatedEvent(serverUrl: string, msg: WebsocketMessage) {
+    let categories;
+
+    try {
+        categories = JSON.parse(msg.data.updatedCategories!);
+        addOrUpdateCategories(serverUrl, categories);
+    } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log('Category WS: handleCategoryUpdatedEvent', e, msg);
+        fetchCategories(serverUrl, msg.broadcast.team_id, false, true);
+    }
+}
+
+export async function handleCategoryDeletedEvent(serverUrl: string, msg: WebsocketMessage) {
+    try {
+        // Delete the Category
+        await deleteCategory(serverUrl, msg.data.category_id);
+
+        // Fetch the categories again as channels will have moved
+        fetchCategories(serverUrl, msg.broadcast.team_id);
```

Comment:
you should prune here too or am I missing something ?

Also don't you already know to which category the channels were moved? or we cannot infer that info?

Reply:
We can - but since we already delete the category, all we need is fetch (this event only fires for one category).